### ISSUE TITLE
repr of universal ABI modules

### ISSUE THREAD

**Issue by antocuni** [hpyproject/hpy#191]
Consider this:
```
$ cd proof-of-concept/
$ python setup.py --hpy-abi=universal build_ext -i
$ python
>>> import pof
>>> pof.__file__
'/home/antocuni/pypy/hpy/proof-of-concept/pof.hpy.so'
>>> pof
<module 'pof' from '/home/antocuni/pypy/hpy/proof-of-concept/pof.py'>
>>> pof.__spec__
ModuleSpec(name='pof', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7fa4ed81c6e0>, origin='/home/antocuni/pypy/hpy/proof-of-concept/pof.py')
```

Note that `pof.__file__` says `pof.hpy.so`, but `repr(pof)` says `pof.py`. I **think** that the repr uses `pof.__spec__.origin` instead of `pof.__file__`.

This is even more obscure because our stub **does** save the new `__spec__` on the just-loaded module, see line 123:
https://github.com/hpyproject/hpy/blob/cc98cac7f623572c4b0096c1767d738b57b4beae/hpy/devel/__init__.py#L113-L127

I think that `importlib` overrides the `__spec__` later.  Probably there are higher-priority issues than this, but nevertheless the current behavior is a bit obscure, since we have a module which claims to be `pof.py` but it has the contents of `pof.hpy.so`.

**Comment by cklein** [comment]
You are right with your first assumption, because the module has a `__spec__` attribute, [_module_repr_from_spec](https://github.com/python/cpython/blob/a7ff6df60c05e1b69fca743573b1e118bebf121d/Lib/importlib/_bootstrap.py#L297-L298) is executed which [returns the origin](https://github.com/python/cpython/blob/a7ff6df60c05e1b69fca743573b1e118bebf121d/Lib/importlib/_bootstrap.py#L591-L592).

At the time when `m.__spec__ = __spec__` is executed, `__spec__.origin` is set to `hello_new.py` which makes to my fairly limited understanding because it's the spec of the module currently being imported, which is `hello_new.py`.
Setting `m.__spec__.origin = ext_filepath` seems to return what you are expecting:

```
>>> hello_new
<module 'hello_new' from '...hello_new.hpy.so'>
```

**Comment by antocuni** [comment]
thank you for investigating into this @cklein. If you want to make a PR I'll be happy to accept it (if you don't feel like I'll just open one next week and I'll give you the credit).

Btw, since I see that you are starting to contribute to hpy, feel free to come to the IRC `#hpy` channel on freenod to say hi, that's where most hpy-related discussions happen.

**Comment by hodgestar** [comment]
Fixed by @cklein in #195.
